### PR TITLE
O3-3434: SDK Command: Remove dependencies from distro.properties

### PR DESCRIPTION
**Pull request by wikumChamith**
## Description of what I changed

This adds a plugin that allows users to manipulate a distro.properties file by removing dependencies.

## Issue I worked on

see https://openmrs.atlassian.net/browse/O3-3434

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean install` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute the above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/RemoveDependency.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            File userDir = new File(System.getProperty("user.dir"));
+            File distroFile = new File(userDir, DistroProperties.DISTRO_FILE_NAME);
+            if (distroFile.exists()) {
+                distro = distroFile.getAbsolutePath();
+            }
+        }
+
+        Properties properties = new Properties();
+
+        if(StringUtils.isNotBlank(distro)) {
```
Comment (dkayiwa) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/RemoveDependency.java:42]: It would be great to always put space after `if`

Comment (wikumChamith) [reply maven-plugin/src/main/java/org/openmrs/maven/plugins/RemoveDependency.java]: fixed it :)

**Comment by wikumChamith** [comment]
@ibacher, @dkayiwa how do you think we should handle exclusions here?

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/RemoveDependency.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.openmrs.maven.plugins.model.DistroProperties;
+
+import java.io.File;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+@Mojo(name = "rm-dependency", requiresProject = false)
```
Comment (ibacher) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/RemoveDependency.java:14]: I'd prefer this was just named `remove`. I don't think `rm-dependency` is very intuitive of a name...

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/RemoveDependency.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    @Override
+    public void executeTask() throws MojoExecutionException, MojoFailureException {
+
+        if (distro == null) {
+            File userDir = new File(System.getProperty("user.dir"));
+            File distroFile = new File(userDir, DistroProperties.DISTRO_FILE_NAME);
+            if (distroFile.exists()) {
+                distro = distroFile.getAbsolutePath();
+            }
+        }
```
Comment (ibacher) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/RemoveDependency.java:38]: So what happens here if there's no distro file in the local directory and none supplied?

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/RemoveDependency.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            property = wizard.promptForMissingValueWithOptions("Enter the property you want to remove",
+                    null, null, properties.keySet().stream().map(prop -> (String) prop).collect(Collectors.toList()));
+        }
+
+        if (StringUtils.isNotBlank(property)) {
+            if (!properties.containsKey(property)) {
+                throw new MojoExecutionException("The property " + property + " was not found in the distro");
+            }
+            properties.remove(property);
+            DistroProperties distroProperties = new DistroProperties(properties);
```
Comment (ibacher) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/RemoveDependency.java:57]: Shouldn't we just move `remove()` to `DistroProperties`? That seems like that removes the need to add `getProperties()` and then create a new `DistroProperties` object, no?

File: integration-tests/pom.xml [hunk 5]
```diff
@@ -42,5 +42,9 @@
             <artifactId>hamcrest-library</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
```
Comment (ibacher) [inline integration-tests/pom.xml:46]: IS this addition actually required? We _shouldn't_ be using mocks in integration tests unless its unavoidable.

**Review by ibacher — CHANGES_REQUESTED** [review]
Thanks @wikumChamith!

**Comment by wikumChamith** [comment]
@ibacher I updated the PR.

**Review by ibacher — APPROVED** [review]
